php/WP_Mock/Functions/Handler.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace WP_Mock\Functions;
+
+use Exception;
+use Mockery\Mock;
+use PHPUnit\Framework\ExpectationFailedException;
+use WP_Mock;
+
+/**
+ * Functions handler.
+ *
+ * This internal handler is meant to handle predefined function calls.
+ *
+ * @see WP_Mock\Functions
+ */
+class Handler
+{
+    /**
+     * Mocked method handlers registered by the test class.
+     *
+     * @var callable[]|callable-string[]
+     */
+    private static array $handlers = [];
+
+    /**
+     * Overrides any existing handlers to set a new callback.
+     *
+     * @param string|callable-string $function function name
+     * @param array<string|object|Mock, callable-string|string>|callable $callback
+     * @return void
+     */
+    public static function registerHandler(string $function, $callback): void
+    {
+        /** @phpstan-ignore-next-line */
+        self::$handlers[$function] = $callback;
+    }
+
+    /**
+     * Handles a mocked function call.
+     *
+     * @param string|callable-string $functionName function name
+     * @param array<mixed> $args function arguments
+     * @return mixed
+     * @throws ExpectationFailedException
+     */
+    public static function handleFunction(string $functionName, array $args = [])
+    {
+        if (self::handlerExists($functionName)) {
```

Comment:
what do you think about using `stattic::` instead of `self::`?

Reply:
I actually just moved this class from its original into a namespace and did not alter its contents